assets/src/api.ts

```diff
@@ -66,6 +67,37 @@ export const apiCall = <T>({
       }
     })
 
+export const checkedApiCall = <T, U>({
+  url,
+  dataStruct,
+  parser,
+  defaultResult,
+  fetchArgs,
+}: {
+  url: string
+  dataStruct: Struct<T, any>
+  parser: (data: T) => U
+  defaultResult?: U
+  fetchArgs?: RequestInit
+}): Promise<U> =>
+  fetch(url, fetchArgs)
+    .then(checkResponseStatus)
+    .then(
+      (response) =>
+        parseJson(response) as Record<string, unknown> & { data: unknown }
+    )
+    .then(({ data: data }) => {
+      assert(data, dataStruct)
```

Comment:
thought: one concern here is the strictness of the `object` check if it will be raising an error. From the [doc](https://docs.superstructjs.org/api-reference/types#object), it seems that if there are extra data properties returned from the backend that this would produce an error. Maybe only fresh in my mind from recently adding a field to the `notificationsData` on the backend, but I do wonder if there is some leniency that is permissible in that or other scenarios.

Reply:
Thinking out loud here, for adding a field to an object (probably the most common change we make to the data coming from the back-end), this means that the steps we would have to do are:

1. Deploy a change to add the optional field to the object in Superstruct.
2. Add the field in the data returned by the backend.
3. Deploy another frontend change to actually use the field and make it required if applicable.

Using `type` instead would allow us to skip the first step, so the process would work much like it does today. Given that extra fields aren't really something I'm worried about (and the data objects get transformed into the actual internal objects anyway, meaning that unknown fields just get dropped) I think that would be a sensible change.